OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -232,6 +232,10 @@ public void onNothingSelected(@NonNull AdapterView<?> parent) {
         final String threadId = getArguments().getString(Router.EXTRA_DISCUSSION_THREAD_ID);
         if(!TextUtils.isEmpty(threadId)){
             router.showCourseDiscussionResponses(context, threadId, courseData);
+
+            // Put discussion_thread_id to null, so that upon recreation of the fragment deep link
+            // is not auto change the screen again.
```

Comment:
// Setting this to null, so that upon recreation of the fragment, relevant activity
// shouldn't be auto-created again (e.g. due to a deep link).